### ISSUE TITLE
Clustering from pre-detected MUA 

### ISSUE THREAD

**Issue by Timothysit** [spyking-circus/spyking-circus#251]
Hi, I want to ask whether it is possible to provide the algorithm with pre-detected MUA spike times along with the filtered data, then sort the spikes using spyking-circus. This is mainly because I am using a spike detection method that is not threshold-based. Thanks!

**Comment by yger** [comment]
What do you mean by pre-detected MUA spikes. The code has a function such that if you are providing your own spike times and labels, it will create the templates for you, and run the template-matching step only. So in a way, you can bypass the clustering. But in your case, this is not what you are asking, you want to simply bypass the detection step, am I correct? So you want to provide a list of spike times, channel_id where you have an event, is that correct?

**Comment by Timothysit** [comment]
Yes! I want to provide a list of both the spike times and the channel_id where I detected the spikes.

**Comment by yger** [comment]
This is not implemented by default in the algorithm. But I could make it as an option, if you are interested, since it should not be too difficult. I guess what the program would need will be a 2D array of spike times (in timesteps) and channel IDs

**Comment by Timothysit** [comment]
Yes, that will be great and I will be a happy to test it out, thanks!

**Comment by abhijithkrishnan93** [comment]
Is this possible. I also would like to provide timings which are detected using another method.
